label_studio/projects/functions/next_task.py

```diff
@@ -1,27 +1,30 @@
 import logging
 from collections import Counter
+from typing import List, Tuple, Union
 
 from core.feature_flags import flag_set
 from core.utils.common import conditional_atomic, db_is_not_sqlite, load_func
 from django.conf import settings
-from django.db.models import BooleanField, Case, Count, Exists, F, Max, OuterRef, Q, Value, When
+from django.db.models import BooleanField, Case, Count, Exists, F, Max, OuterRef, Q, QuerySet, Value, When
 from django.db.models.fields import DecimalField
 from projects.functions.stream_history import add_stream_history
+from projects.models import Project
 from tasks.models import Annotation, Task
+from users.models import User
 
 logger = logging.getLogger(__name__)
 
 get_tasks_agreement_queryset = load_func(settings.GET_TASKS_AGREEMENT_QUERYSET)
 
 
-def get_next_task_logging_level(user):
+def get_next_task_logging_level(user: User):
     level = logging.DEBUG
     if flag_set('fflag_fix_back_dev_4185_next_task_additional_logging_long', user=user):
         level = logging.INFO
     return level
 
 
-def _get_random_unlocked(task_query, user, upper_limit=None):
+def _get_random_unlocked(task_query: QuerySet[Task], user: User, upper_limit=None) -> Union[Task, None]:
```

Comment:
`upper_limit` appears unused, can we remove it?

Reply:
There was a usage I found where we were intending to use it, it is passed from another call site, but in the function unused. I'm going to leave this one, as the removal of the usage seems like it would cause an issue in typing if I changed the parameter set. Let's take note of this function `_get_random_unlocked` for future to review the expectations of its' usages.